src/hooks/use-filtered-market-coins.hook.ts

```diff
@@ -0,0 +1,63 @@
+import { useMemo, useState, useCallback } from 'react';
+import { useDispatch } from 'react-redux';
+
+import { MarketCoinsSortFieldEnum } from '../enums/market-coins-sort-field.enum';
+import { selectSortValue } from '../store/market/market-actions';
+import {
+  useFavouriteTokens,
+  useMarketTopCoinsWithoutTez,
+  useSortFieldSelector
+} from '../store/market/market-selectors';
+import { isString } from '../utils/is-string';
+import { sortMarketCoins } from '../utils/market.util';
+import { MarketCoin } from './../store/market/market.interfaces';
+
+export const useFilterdMarketCoins = () => {
+  const dispatch = useDispatch();
+  const tokens = useMarketTopCoinsWithoutTez();
+  const sortFiled = useSortFieldSelector();
+  const favouriteTokens = useFavouriteTokens();
+
+  const inputTypeIndexDefault = favouriteTokens.length === 0 ? 0 : 1;
+
+  const [searchValue, setSearchValue] = useState<string>();
+  const [inputTypeIndex, setInputTypeIndex] = useState<number>(inputTypeIndexDefault);
```

Comment:
I do not understand what does `inputTypeIndex` means. As I understood it used as SegmantControl index - but name is meaningless